system/Database/BaseResult.php

```diff
@@ -260,7 +260,7 @@ public function getResultObject(): array
      * @param         string                           $type The type of result object. 'array', 'object' or class name.
      * @phpstan-param class-string<T>|'array'|'object' $type
      *
-     * @return         array|object|stdClass|null
+     * @return         array|object|stdClass|null|string
```

Comment:
Oh, yes, `$n` might be a column name.
So the return value might be `int` or `float`.